File: algorithms/26. Remove Duplicates from Sorted Array.cpp
#include <vector>
#include <set>
using namespace std;

int removeDuplicates(vector<int>& a) {
    int n = a.size();
    int l = 0;
    for (int r = 0; r < n; r++) {
        if (r == 0 || a[r - 1] != a[r]) {
            a[l] = a[r];
            l++;
        }
    }
    return l;
}

int removeDuplicates22222222(vector<int>& a) {
    set<int> s(a.begin(), a.end());
    int i = 0;
    for (int x : s) {
        a[i] = x;
        i++;
    }
    return i;
}

File: algorithms/26. Remove Duplicates from Sorted Array.js
var removeDuplicates = function (a) {
    let n = a.length

    let l = 0, r = 0
    for (let r = 0; r < n; r++) {
        if (a[r - 1] !== a[r]) {
            a[l] = a[r]
            l++
        }
    }
    return l
};

var removeDuplicates22222222 = function (a) {
    let set = new Set(a)

    let i = 0
    for (let x of set) {
        a[i] = x
        i++
    }
    return i
};
